clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -399,14 +400,45 @@ public void testHeartbeatState() {
             new ConsumerGroupHeartbeatResponseData.Assignment();
         assignmentTopic1.setTopicPartitions(Collections.singletonList(tpTopic1));
         ConsumerGroupHeartbeatResponse rs1 = new ConsumerGroupHeartbeatResponse(new ConsumerGroupHeartbeatResponseData()
-                .setHeartbeatIntervalMs(DEFAULT_HEARTBEAT_INTERVAL_MS)
-                .setMemberId(memberId)
-                .setMemberEpoch(1)
-                .setAssignment(assignmentTopic1));
+            .setHeartbeatIntervalMs(DEFAULT_HEARTBEAT_INTERVAL_MS)
+            .setMemberId(memberId)
+            .setMemberEpoch(1)
+            .setAssignment(assignmentTopic1));
         membershipManager.onHeartbeatResponseReceived(rs1.data());
         assertEquals(MemberState.RECONCILING, membershipManager.state());
     }
 
+    @Test
+    public void testEnsureLeaveGroupWhenPollTimerExpires() {
+        membershipManager.transitionToJoining();
```

Comment:
You could consider using `membershipManager.transitionToFenced(); // And indirect way of moving to JOINING state` instead of changing the visibility of prod code

Reply:
it is not possible to transition to fenced from unsubscribed

Reply:
you are 100% correct.  I submit a patch for this specific test.  Thanks for catching this.